aidial_adapter_openai/utils/streaming.py

```diff
@@ -51,42 +51,45 @@ def build_chunk(
 
 
 async def generate_stream(
-    prompt_tokens: int,
-    stream: AsyncIterator[dict],
-    tokenizer: Tokenizer,
+    *,
+    get_prompt_tokens: Callable[[], int],
+    tokenize: Callable[[str], int],
     deployment: str,
     discarded_messages: Optional[list[int]],
+    stream: AsyncIterator[dict],
 ) -> AsyncIterator[dict]:
+
     last_chunk, temp_chunk = None, None
     stream_finished = False
+    total_content = ""
+
+    def finalize_last_chunk(chunk: dict) -> None:
```

Comment:
Let's avoid any functions with hidden changes of the object
When I see this
```python
async for chunk in stream:
    finalize_last_chunk(chunk)
```
I don't know from first glance, that chunk was modified, on contrary to


```python
async for chunk in stream:
    chunk = finalize_last_chunk(chunk)
```